Assets/Scripts/Guests/Guest.cs

```diff
@@ -175,8 +345,13 @@ private void HandleSinking(bool floodedOrSunk)
 	/// </summary>
 	private void BeginLeaving()
 	{
-		transform.position = ExitPoint;
-		sprite.sortingLayerID = SortingLayer.NameToID("GuestBehindHotel");
+		if(currentRoom != null)
+		{
+			ChangeRoom(null);
+			transform.position = ExitPoint;

```

Comment:
does this imply that the guest was at the door position, or is that not a requirement?

Reply:
If currentRoom is null, guest is in the backrooms, so no need to teleport them to the exitpoint (right staircase). Example: guests have not arrived to the hotel yet, and you hit the Make All Guests Leave button. If this check is missing, the guests will teleport across the map.